lib/slack-ruby-bot/hooks/hook_support.rb

```diff
@@ -0,0 +1,39 @@
+module SlackRubyBot
+  module Hooks
+    module HookSupport
+      def self.included(base)
+        base.cattr_accessor :hook_blocks
+
+        base.extend(ClassMethods)
+      end
+
+      module ClassMethods
+        def on(event_name, &block)
+          self.hook_blocks ||= {}
+
+          self.hook_blocks[event_name] ||= []
+          self.hook_blocks[event_name] << block
+        end
+      end
+
+      # Instance stuff
+      attr_writer :hooks
```

Comment:
Is this ever assigned? Maybe it should be an `attr_reader` and marked `private`?